File: 001 - Multiples of 3 and 5/001 - Multiples_of_3_and_5.jl
#=
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
Find the sum of all the multiples of 3 or 5 below 1000

=#

function multiplesOf3and5(n::Int)
    currSum = 0
    for x = 1:n-1
        if x % 3 == 0 || x % 5 == 0
            currSum += x
        end
    end
    return currSum
end

println(multiplesOf3and5(1000))

File: haskell/001 - Multiples of 3 and 5/001 - Multiples_of_3_and_5.hs
{-
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
Find the sum of all the multiples of 3 or 5 below 1000
-}

multiplesOf3and5 :: Int -> Int
multiplesOf3and5 n = sum [x | x <- [1..n-1], x `mod` 3 == 0 || x `mod` 5 == 0]

main :: IO ()
main = putStrLn $ show (multiplesOf3and5 1000)

